pyproject.toml

```diff
@@ -29,7 +29,7 @@ requires-python = '>=3.11'
 dependencies = [
   "numpy>=1.26.0",
   "python-dateutil>=2.8.2",
-  "tzdata>=2023.3"
+  "tzdata>=2023.3; platform_system == 'Windows'",
```

Comment:
With tests seeming to be fine with no minimum here (ref: https://github.com/pandas-dev/pandas/issues/63264#issuecomment-3629049941), I'm wondering if we should remove the lower bound.